components/posts/posts.php

```diff
@@ -132,25 +148,23 @@ function mm_posts( $args ) {
 
 	ob_start(); ?>
 
-	<div class="<?php echo esc_attr( $mm_classes ); ?>">
+	<?php do_action( 'mm_posts_before', $query, $context, $args ); ?>
```

Comment:
I think relocating this hook might cause problems, and it means that now `mm_posts_before` fires outside the wrapper div but `mm_posts_after` fires inside, but I agree that we need hooks before and after that are outside the div. What if you rename `mm_posts_before` and `mm_posts_after` to `mm_posts_before_inside` and `mm_posts_after_inside`, then you can have `mm_posts_before` and `mm_posts_after` be outside the wrapper div. I think the only thing you'd need to switch is the pagination output from `mm_posts_after` to `mm_posts_after_inside` and the masonry sizer to `mm_posts_before_inside` in `mm_posts_register_default_hooks`.